src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+
 public class WorkWithFile {
+    private static final String COMMA_DELIMITER = ",";
+    private static final String LINE_DELIMITER = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String report = formReport(readFile(fromFileName));
+        writeToFile(toFileName, report);
+    }
+
+    private String[] readFile(String fromFileName) {
+        File sourceFile = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        String[] data;
+        try {
+            String fileContent = Files.readString(sourceFile.toPath());
+            data = fileContent.toLowerCase().split(LINE_DELIMITER);
```

Comment:
Using `toLowerCase()` on the entire file content may not be necessary and can be misleading. If the purpose is to normalize the operation types (e.g., 'supply', 'buy'), it should be applied specifically to those parts of the data, not the whole content.